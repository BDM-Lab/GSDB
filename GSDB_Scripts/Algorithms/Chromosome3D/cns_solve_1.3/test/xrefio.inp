remarks file test/xrefio.inp
remarks  tests reflection input/output


xray
   declare name=fobs type=complex domain=reciprocal end
   declare name=fpart domain=reciprocal type=complex end
   declare name=fcalc domain=reciprocal type=complex end
   declare name=fom domain=reciprocal type=real end
   declare name=weight domain=reciprocal type=real end
   declare name=sigma domain=reciprocal type=real end
   declare name=test domain=reciprocal type=integer end

  refl
   nref=200

   index 1 2 3 fobs 40. 50.  fcalc 30. 20.  test 1 
 end

 write refl sele=( all ) end

 declare name=deri1 type=complex domain=reciprocal end

 do (deri1=40) ( all )

 write refl sele=( all ) end

 undeclare name=deri1 domain=reciprocal end

 write refl sele=( all ) end

 do (weight=0.9) ( all )

 write refl sele=( all ) end

 do (weight=1.0) ( all )

 write refl sele=( all ) end

 declare name=uuu type=real domain=reciprocal end
 declare name=vvv type=integer domain=reciprocal end
 declare name=ttt type=complex domain=reciprocal end

 write refl sele=( all ) end

 do (uuu=3) ( all )

 write refl sele=( all ) end

 do (vvv=4) ( all )
 write refl sele=( all ) end

 do (ttt=4+i * 4 ) ( all )

 write refl sele=( all ) end

 undeclare name=uuu domain=reciprocal end
 undeclare name=vvv domain=reciprocal end
 undeclare name=ttt domain=reciprocal end

 refl
   nref=1
   
   declare name=uuu type=real domain=reciprocal end
   declare name=vvv type=integer domain=reciprocal end
   declare name=ttt type=complex domain=reciprocal end
   index 1 2 3 fobs 40. 30. uuu 3 vvv 4 ttt 10 10 test 2 weight 2
 end

 write refl sele=( all ) end

 write refl fobs test sele=( all ) end

 refl
   index 1 2 3 fobs 40 30  fcalc 40. 40
 end

 write refl fobs fcalc sele=( all ) end

 
 refl
   reset
 end

 a=100 b=100 c=100  

  evaluate ($lr=10.)
  evaluate ($hr=9.9) 
  binresolution $lr $hr 
  mapresolution $hr
  tselection=( $hr <= d <= $lr ) 

 generate 9.9 10. 

 do (fobs=combine(d,0)) ( all ) 
 undeclare name=uuu domain=reciprocal end
 
 declare name=uuu type=complex domain=reciprocal end

 do (uuu=combine(d*2,0)) ( all )

 do (weight=d*3) ( all )

 do (sigma=d*4) ( all )

  evaluate ($lr=10.)
  evaluate ($hr=9.) 
  binresolution $lr $hr 
  mapresolution $hr
  tselection=( $hr <= d <= $lr ) 

 nref=1000
 generate 9. 10. 

 show max (abs(amplitude(fobs)-d)) ( 9.9 <= d <=10. ) 
 show max (amplitude(fobs)) ( 9 <= d< 9.9) 


 show max (abs(amplitude(uuu)-d*2)) ( 9.9 <= d <=10. ) 
 show max (amplitude(uuu)) ( 9 <= d< 9.9) 

 show max (abs(weight-d*3)) ( 9.9 <= d <=10. ) 
 show max (weight) ( 9 <= d< 9.9) 

 show max (abs(sigma-d*4)) ( 9.9 <= d <=10. ) 
 show max (sigma) ( 9 <= d< 9.9) 
end

stop




