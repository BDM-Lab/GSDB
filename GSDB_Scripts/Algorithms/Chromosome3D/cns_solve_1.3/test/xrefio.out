          ============================================================
          |                                                          |
          |            Crystallography & NMR System (CNS)            |
          |                         CNSsolve                         |
          |                                                          |
          ============================================================
           Version: 1.3
           Status: General release
          ============================================================
           Written by: A.T.Brunger, P.D.Adams, G.M.Clore, W.L.DeLano,
                       P.Gros, R.W.Grosse-Kunstleve,J.-S.Jiang,J.M.Krahn,
                       J.Kuszewski, M.Nilges, N.S.Pannu, R.J.Read,
                       L.M.Rice, G.F.Schroeder, T.Simonson, G.L.Warren.
           Copyright (c) 1997-2010 Yale University
          ============================================================
           Running on machine: lewis4-r710-login-node223 (x86_64/Linux,64-bit)
                               with 16 threads
           Program started by: oeow39
           Program started at: 16:08:33 on 05-Jul-2018
          ============================================================

 FFT3C: Using FFTPACK4.1

 CNSsolve>remarks file test/xrefio.inp 
 CNSsolve>remarks  tests reflection input/output 
 CNSsolve> 
 CNSsolve> 
 CNSsolve>xray 
 XRAY>   declare name=fobs type=complex domain=reciprocal end 
 XDECLARE: Object FOBS has been declared.
 XRAY>   declare name=fpart domain=reciprocal type=complex end 
 XDECLARE: Object FPART has been declared.
 XRAY>   declare name=fcalc domain=reciprocal type=complex end 
 XDECLARE: Object FCALC has been declared.
 XRAY>   declare name=fom domain=reciprocal type=real end 
 XDECLARE: Object FOM has been declared.
 XRAY>   declare name=weight domain=reciprocal type=real end 
 XDECLARE: Object WEIGHT has been declared.
 XRAY>   declare name=sigma domain=reciprocal type=real end 
 XDECLARE: Object SIGMA has been declared.
 XRAY>   declare name=test domain=reciprocal type=integer end 
 XDECLARE: Object TEST has been declared.
 XRAY> 
 XRAY>  refl 
 REFLection>   nref=200 
 REFLection> 
 REFLection>   index 1 2 3 fobs 40. 50.  fcalc 30. 20.  test 1 
 XSFAL: allocating space for complex reciprocal space object.
 XSFAL: allocating space for complex reciprocal space object.
 XSFAL: allocating space for integer reciprocal space object.
 XRRR2:        1 new h,k,l indices have been added.
 XRAY> 
 XRAY> write refl sele=( all ) end 
 Total of         1 structure factor elements were selected.
 NREFlection=         1
 ANOMalous=FALSe { equiv. to HERMitian=TRUE}
 DECLare NAME=FOBS                   DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=FCALC                  DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=TEST                   DOMAin=RECIprocal   TYPE=INTE END
 INDE     1    2    3 FOBS=    40.000    50.000 FCALC=    30.000    20.000
                   TEST=         1
 XRAY> 
 XRAY> declare name=deri1 type=complex domain=reciprocal end 
 XDECLARE: Object DERI1 has been declared.
 XRAY> 
 XRAY> do (deri1=40) ( all ) 
 XSFAL: allocating space for complex reciprocal space object.
 Total of         1 structure factor elements were selected.
 XRAY> 
 XRAY> write refl sele=( all ) end 
 Total of         1 structure factor elements were selected.
 NREFlection=         1
 ANOMalous=FALSe { equiv. to HERMitian=TRUE}
 DECLare NAME=FOBS                   DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=FCALC                  DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=TEST                   DOMAin=RECIprocal   TYPE=INTE END
 DECLare NAME=DERI1                  DOMAin=RECIprocal   TYPE=COMP END
 INDE     1    2    3 FOBS=    40.000    50.000 FCALC=    30.000    20.000
                   TEST=         1 DERI1=    40.000     0.000
 XRAY> 
 XRAY> undeclare name=deri1 domain=reciprocal end 
 Object DERI1                has been undeclared.
 XRAY> 
 XRAY> write refl sele=( all ) end 
 Total of         1 structure factor elements were selected.
 NREFlection=         1
 ANOMalous=FALSe { equiv. to HERMitian=TRUE}
 DECLare NAME=FOBS                   DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=FCALC                  DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=TEST                   DOMAin=RECIprocal   TYPE=INTE END
 INDE     1    2    3 FOBS=    40.000    50.000 FCALC=    30.000    20.000
                   TEST=         1
 XRAY> 
 XRAY> do (weight=0.9) ( all ) 
 XSFAL: allocating space for real reciprocal space object.
 Total of         1 structure factor elements were selected.
 XRAY> 
 XRAY> write refl sele=( all ) end 
 Total of         1 structure factor elements were selected.
 NREFlection=         1
 ANOMalous=FALSe { equiv. to HERMitian=TRUE}
 DECLare NAME=FOBS                   DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=FCALC                  DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=WEIGHT                 DOMAin=RECIprocal   TYPE=REAL END
 DECLare NAME=TEST                   DOMAin=RECIprocal   TYPE=INTE END
 INDE     1    2    3 FOBS=    40.000    50.000 FCALC=    30.000    20.000
                   WEIGHT=     0.900 TEST=         1
 XRAY> 
 XRAY> do (weight=1.0) ( all ) 
 Total of         1 structure factor elements were selected.
 XRAY> 
 XRAY> write refl sele=( all ) end 
 Total of         1 structure factor elements were selected.
 NREFlection=         1
 ANOMalous=FALSe { equiv. to HERMitian=TRUE}
 DECLare NAME=FOBS                   DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=FCALC                  DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=WEIGHT                 DOMAin=RECIprocal   TYPE=REAL END
 DECLare NAME=TEST                   DOMAin=RECIprocal   TYPE=INTE END
 INDE     1    2    3 FOBS=    40.000    50.000 FCALC=    30.000    20.000
                   WEIGHT=     1.000 TEST=         1
 XRAY> 
 XRAY> declare name=uuu type=real domain=reciprocal end 
 XDECLARE: Object UUU has been declared.
 XRAY> declare name=vvv type=integer domain=reciprocal end 
 XDECLARE: Object VVV has been declared.
 XRAY> declare name=ttt type=complex domain=reciprocal end 
 XDECLARE: Object TTT has been declared.
 XRAY> 
 XRAY> write refl sele=( all ) end 
 Total of         1 structure factor elements were selected.
 NREFlection=         1
 ANOMalous=FALSe { equiv. to HERMitian=TRUE}
 DECLare NAME=FOBS                   DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=FCALC                  DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=WEIGHT                 DOMAin=RECIprocal   TYPE=REAL END
 DECLare NAME=TEST                   DOMAin=RECIprocal   TYPE=INTE END
 INDE     1    2    3 FOBS=    40.000    50.000 FCALC=    30.000    20.000
                   WEIGHT=     1.000 TEST=         1
 XRAY> 
 XRAY> do (uuu=3) ( all ) 
 XSFAL: allocating space for real reciprocal space object.
 Total of         1 structure factor elements were selected.
 XRAY> 
 XRAY> write refl sele=( all ) end 
 Total of         1 structure factor elements were selected.
 NREFlection=         1
 ANOMalous=FALSe { equiv. to HERMitian=TRUE}
 DECLare NAME=FOBS                   DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=FCALC                  DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=WEIGHT                 DOMAin=RECIprocal   TYPE=REAL END
 DECLare NAME=TEST                   DOMAin=RECIprocal   TYPE=INTE END
 DECLare NAME=UUU                    DOMAin=RECIprocal   TYPE=REAL END
 INDE     1    2    3 FOBS=    40.000    50.000 FCALC=    30.000    20.000
                   WEIGHT=     1.000 TEST=         1 UUU=     3.000
 XRAY> 
 XRAY> do (vvv=4) ( all ) 
 XSFAL: allocating space for integer reciprocal space object.
 Total of         1 structure factor elements were selected.
 XRAY> write refl sele=( all ) end 
 Total of         1 structure factor elements were selected.
 NREFlection=         1
 ANOMalous=FALSe { equiv. to HERMitian=TRUE}
 DECLare NAME=FOBS                   DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=FCALC                  DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=WEIGHT                 DOMAin=RECIprocal   TYPE=REAL END
 DECLare NAME=TEST                   DOMAin=RECIprocal   TYPE=INTE END
 DECLare NAME=UUU                    DOMAin=RECIprocal   TYPE=REAL END
 DECLare NAME=VVV                    DOMAin=RECIprocal   TYPE=INTE END
 INDE     1    2    3 FOBS=    40.000    50.000 FCALC=    30.000    20.000
                   WEIGHT=     1.000 TEST=         1 UUU=     3.000
                   VVV=         4
 XRAY> 
 XRAY> do (ttt=4+i * 4 ) ( all ) 
 XSFAL: allocating space for complex reciprocal space object.
 Total of         1 structure factor elements were selected.
 XRAY> 
 XRAY> write refl sele=( all ) end 
 Total of         1 structure factor elements were selected.
 NREFlection=         1
 ANOMalous=FALSe { equiv. to HERMitian=TRUE}
 DECLare NAME=FOBS                   DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=FCALC                  DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=WEIGHT                 DOMAin=RECIprocal   TYPE=REAL END
 DECLare NAME=TEST                   DOMAin=RECIprocal   TYPE=INTE END
 DECLare NAME=UUU                    DOMAin=RECIprocal   TYPE=REAL END
 DECLare NAME=VVV                    DOMAin=RECIprocal   TYPE=INTE END
 DECLare NAME=TTT                    DOMAin=RECIprocal   TYPE=COMP END
 INDE     1    2    3 FOBS=    40.000    50.000 FCALC=    30.000    20.000
                   WEIGHT=     1.000 TEST=         1 UUU=     3.000
                   VVV=         4 TTT=     5.657    45.000
 XRAY> 
 XRAY> undeclare name=uuu domain=reciprocal end 
 Object UUU                  has been undeclared.
 XRAY> undeclare name=vvv domain=reciprocal end 
 Object VVV                  has been undeclared.
 XRAY> undeclare name=ttt domain=reciprocal end 
 Object TTT                  has been undeclared.
 XRAY> 
 XRAY> refl 
 REFLection>   nref=1 
 REFLection: data will be merged with existing data.
 REFLection> 
 REFLection>   declare name=uuu type=real domain=reciprocal end 
 XDECLARE: Object UUU has been declared.
 REFLection>   declare name=vvv type=integer domain=reciprocal end 
 XDECLARE: Object VVV has been declared.
 REFLection>   declare name=ttt type=complex domain=reciprocal end 
 XDECLARE: Object TTT has been declared.
 REFLection>   index 1 2 3 fobs 40. 30. uuu 3 vvv 4 ttt 10 10 test 2 weight 2 
 XSFAL: allocating space for real reciprocal space object.
 XSFAL: allocating space for integer reciprocal space object.
 XSFAL: allocating space for complex reciprocal space object.
 XRRR2:        0 new h,k,l indices have been added.
 XRAY> 
 XRAY> write refl sele=( all ) end 
 Total of         1 structure factor elements were selected.
 NREFlection=         1
 ANOMalous=FALSe { equiv. to HERMitian=TRUE}
 DECLare NAME=FOBS                   DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=FCALC                  DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=WEIGHT                 DOMAin=RECIprocal   TYPE=REAL END
 DECLare NAME=TEST                   DOMAin=RECIprocal   TYPE=INTE END
 DECLare NAME=UUU                    DOMAin=RECIprocal   TYPE=REAL END
 DECLare NAME=VVV                    DOMAin=RECIprocal   TYPE=INTE END
 DECLare NAME=TTT                    DOMAin=RECIprocal   TYPE=COMP END
 INDE     1    2    3 FOBS=    40.000    30.000 FCALC=    30.000    20.000
                   WEIGHT=     2.000 TEST=         2 UUU=     3.000
                   VVV=         4 TTT=    10.000    10.000
 XRAY> 
 XRAY> write refl fobs test sele=( all ) end 
 Total of         1 structure factor elements were selected.
 NREFlection=         1
 ANOMalous=FALSe { equiv. to HERMitian=TRUE}
 DECLare NAME=FOBS                   DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=TEST                   DOMAin=RECIprocal   TYPE=INTE END
 INDE     1    2    3 FOBS=    40.000    30.000 TEST=         2
 XRAY> 
 XRAY> refl 
 REFLection>   index 1 2 3 fobs 40 30  fcalc 40. 40 
 REFLection: data will be merged with existing data.
 XRRR2:        0 new h,k,l indices have been added.
 XRAY> 
 XRAY> write refl fobs fcalc sele=( all ) end 
 Total of         1 structure factor elements were selected.
 NREFlection=         1
 ANOMalous=FALSe { equiv. to HERMitian=TRUE}
 DECLare NAME=FOBS                   DOMAin=RECIprocal   TYPE=COMP END
 DECLare NAME=FCALC                  DOMAin=RECIprocal   TYPE=COMP END
 INDE     1    2    3 FOBS=    40.000    30.000 FCALC=    40.000    40.000
 XRAY> 
 XRAY> 
 XRAY> refl 
 REFLection>   reset 
 REFLection: data will be merged with existing data.
 XRRR: all existing reciprocal data will be scratched.
 REFLection> end 
 XRRR2:        0 new h,k,l indices have been added.
 XRAY> 
 XRAY> a=100 b=100 c=100 
 XRAY> 
 XRAY>  evaluate ($lr=10.) 
 EVALUATE: symbol $LR set to    10.0000     (real)
 XRAY>  evaluate ($hr=9.9) 
 EVALUATE: symbol $HR set to    9.90000     (real)
 XRAY>  binresolution $lr $hr 
 XRAY>  mapresolution $hr 
 XRAY>  tselection=( $hr <= d <= $lr ) 
 XRAY> 
 XRAY> generate 9.9 10. 
 XGENER: generating reflections to produce a full set
         for the specified resolution range.
 XGENE2:      123 new reflections have been generated.
 XRAY> 
 XRAY> do (fobs=combine(d,0)) ( all ) 
 XSFAL: allocating space for complex reciprocal space object.
 Total of       123 structure factor elements were selected.
 XRAY> undeclare name=uuu domain=reciprocal end 
 Object UUU                  has been undeclared.
 XRAY> 
 XRAY> declare name=uuu type=complex domain=reciprocal end 
 XDECLARE: Object UUU has been declared.
 XRAY> 
 XRAY> do (uuu=combine(d*2,0)) ( all ) 
 XSFAL: allocating space for complex reciprocal space object.
 Total of       123 structure factor elements were selected.
 XRAY> 
 XRAY> do (weight=d*3) ( all ) 
 XSFAL: allocating space for real reciprocal space object.
 Total of       123 structure factor elements were selected.
 XRAY> 
 XRAY> do (sigma=d*4) ( all ) 
 XSFAL: allocating space for real reciprocal space object.
 Total of       123 structure factor elements were selected.
 XRAY> 
 XRAY>  evaluate ($lr=10.) 
 EVALUATE: symbol $LR set to    10.0000     (real)
 XRAY>  evaluate ($hr=9.) 
 EVALUATE: symbol $HR set to    9.00000     (real)
 XRAY>  binresolution $lr $hr 
 XRAY>  mapresolution $hr 
 XRAY>  tselection=( $hr <= d <= $lr ) 
 XRAY> 
 XRAY> nref=1000 
 XRAY: allocating space for    1000 reflections.
 XRAY> generate 9. 10. 
 XGENER: existing reflections will be conserved
         appending new reflections to produce
         a full set for the specified resolution range
 XGENE2:      679 new reflections have been generated.
 XRAY> 
 XRAY> show max (abs(amplitude(fobs)-d)) ( 9.9 <= d <=10. ) 
 Maximum of       123 elements =                 0.0000
 XRAY> show max (amplitude(fobs)) ( 9 <= d< 9.9) 
 Maximum of       679 elements =                 0.0000
 XRAY> 
 XRAY> 
 XRAY> show max (abs(amplitude(uuu)-d*2)) ( 9.9 <= d <=10. ) 
 Maximum of       123 elements =                 0.0000
 XRAY> show max (amplitude(uuu)) ( 9 <= d< 9.9) 
 Maximum of       679 elements =                 0.0000
 XRAY> 
 XRAY> show max (abs(weight-d*3)) ( 9.9 <= d <=10. ) 
 Maximum of       123 elements =                 0.0000
 XRAY> show max (weight) ( 9 <= d< 9.9) 
 Maximum of       679 elements =                 0.0000
 XRAY> 
 XRAY> show max (abs(sigma-d*4)) ( 9.9 <= d <=10. ) 
 Maximum of       123 elements =                 0.0000
 XRAY> show max (sigma) ( 9 <= d< 9.9) 
 Maximum of       679 elements =                 0.0000
 XRAY>end 
 CNSsolve> 
 CNSsolve>stop 
          ============================================================
           Maximum dynamic memory allocation:     8405000 bytes
           Maximum dynamic memory overhead:           512 bytes
           Program started at: 16:08:33 on 05-Jul-2018
           Program stopped at: 16:08:33 on 05-Jul-2018
           CPU time used:       0.0094 seconds
          ============================================================
